src/transformers/pipelines/base.py

```diff
@@ -1035,10 +1036,23 @@ def forward(self, model_inputs, **forward_params):
     def get_iterator(
         self, inputs, num_workers: int, batch_size: int, preprocess_params, forward_params, postprocess_params
     ):
+        try:
+            n = len(inputs)
+        except TypeError:
+            # Iterator
+            n = None
+        if n is not None:
+            dataset = PipelineDataset(inputs, self.preprocess, preprocess_params)
+        else:
+            if num_workers > 1:
+                logger.warning(
+                    "For iterable dataset using num_workers>1 is likely to result in errors since everything is iterable, setting `num_workers=1` to guarantee correctness."
```

Comment:
Please respect the 119 char limits here.

Reply:
Why doesn't the quality check detects that ?

Reply:
I modified, but I expect black and formatting tools to do that automatically :(

Reply:
No they only treat the code, not the strings and docstrings, which is annoying.
(I don't really care if it goes over the limit a little bit, but if we don't see the diff properly on GitHub it's really too long and inconvenient ;-) )

Reply:
It is indeed annoying, anyway fixed it.